src/Compilers/Core/Portable/Syntax/SyntaxDiffer.cs

```diff
@@ -568,6 +582,37 @@ private void RecordChange(ChangeRecord change)
             _changes.Add(change);
         }
 
+        private void RecordChange(TextChangeRange textChangeRange, SyntaxNodeOrToken removedNode, SyntaxNodeOrToken insertedNode)
+        {
+            if (_changes.Count > 0)
+            {
+                var last = _changes[_changes.Count - 1];
+                if (last.Range.Span.End == textChangeRange.Span.Start)
+                {
+                    // merge changes...
+                    last.OldNodes?.Enqueue(removedNode);
```

Comment:
Yes, the implementation used null for empty prior to my change and I didn't change that behavior.